code/modules/modular_computers/file_system/programs/statusdisplay.dm

```diff
@@ -15,36 +15,76 @@
 	var/upper_text
 	var/lower_text
 
-/datum/computer_file/program/status/proc/SendSignal()
+/**
+ * Post status display radio packet.
+ * Arguments:
+ * * command - the status display command
+ * * data1 - the data1 value, as defined by status displays
+ * * data2 - the data2 value, as defined by status displays
+ */
+/datum/computer_file/program/status/proc/post_status(command, data1, data2)
 	var/datum/radio_frequency/frequency = SSradio.return_frequency(FREQ_STATUS_DISPLAYS)
-
 	if(!frequency)
 		return
 
-	var/datum/signal/status_signal = new(list("command" = "message"))
+	var/datum/signal/status_signal = new(list("command" = command))
+	switch(command)
+		if("message")
+			status_signal.data["msg1"] = data1
+			status_signal.data["msg2"] = data2
+		if("alert")
+			status_signal.data["picture_state"] = data1
+
+	frequency.post_signal(src, status_signal)
 
-	status_signal.data["msg1"] = reject_bad_text(upper_text || "", MAX_STATUS_LINE_LENGTH)
-	status_signal.data["msg2"] = reject_bad_text(lower_text || "", MAX_STATUS_LINE_LENGTH)
+/**
+ * Post a message to status displays
+ * Arguments:
+ * * upper - Top text
+ * * lower - Bottom text
+ */
+/datum/computer_file/program/status/proc/post_message(upper, lower)
+	post_status("message", upper, lower)
 
-	frequency.post_signal(computer, status_signal)
+/**
+ * Post a picture to status displays
+ * Arguments:
+ * * picture - The picture name
+ */
+/datum/computer_file/program/status/proc/post_picture(picture)
+	var/static/obj/machinery/computer/communications/comm_prototype = /obj/machinery/computer/communications
 
-/datum/computer_file/program/status/proc/SetText(position, text)
-	switch(position)
-		if("upper")
-			upper_text = text
-		if("lower")
-			lower_text = text
+	if (!(picture in initial(comm_prototype.approved_status_pictures)))
+		return
+	if(picture in initial(comm_prototype.state_status_pictures))
+		post_status(picture)
```

Comment:
Why are you using `initial` here?

Reply:
I would otherwise have to locate a communications console, which would be a waste of time, as these are static lists.

Reply:
`initial` is used to get a list or var to it's original value when before the code is run.  See:

https://www.byond.com/docs/ref/#/proc/initial

I don't see the list changing anywhere, which means it always remains the same, making `initial` redundant.

Reply:
comm_prototype isn't an instance, it's a type path. I'm using the trick of grabbing defaults/static values from a type path.

Reply:
Ok sounds good.